addons/api/addon/models/host-set.js

```diff
@@ -71,23 +71,12 @@ export default class HostSetModel extends GeneratedHostSetModel {
    * Host IDs are read-only under normal circumstances.  But these can
    * be persisted via a dedicated call to `addHosts()`.
    */
-  @fragmentArray('fragment-string', {
-    readOnly: true,
-    emptyArrayIfMissing: true,
-  })
-  host_ids;
+  @attr({ readOnly: true, emptyArrayIfMissing: true }) host_ids;
 
-  @fragmentArray('fragment-string', {
-    emptyArrayIfMissing: true,
```

Comment:
@randallmorey Do you know the order of transforms and serializers? 

Does it go transform --> normalizer --> model and then model --> serializer --> transform?

I ask because we removed the `emptyArrayIfMissing` because I assume the transformer handles it for `string-array` but a serializer test was broken because the normalizer wasn't aware of the transform

Reply:
It's possible that if a field is wholly absent, a transform isn't activated.  However, [normalisation of empty arrays based on that attribute option are independent of the transform](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/serializers/application.js#L191).  It's a judgement call to determine if you need/want an empty array.

Reply:
Alright makes sense, I added back the attribute option as I do think it makes sense to have an empty array